nixos/modules/services/networking/kresd.nix

```diff
@@ -2,6 +2,13 @@
 let
   cfg = config.services.kresd;
 
+  # TODO: partial override of defaults doesn't work,
+  # but perhaps it will be OK as empty anyway?
+  settings_default = { };
```

Comment:
You want to move them to config or write something like the following to config.

```
config.services.knot-resolver.settings = options.services.knot-resolver.settings.default;
```